teslajsonpy/homeassistant/homelink.py

```diff
@@ -0,0 +1,90 @@
+"""
+
+from teslajsonpy.exceptions import HomelinkError
+from teslajsonpy.homeassistant.vehicle import VehicleDevice
+
+
+class TriggerHomelink(VehicleDevice):
+    """Home-Assistant class for trigger homelink of Tesla vehicles."""
+
+    def __init__(self, data, controller):
+        """Initialize the trigger homelink for the vehicle.
+
+        Parameters
+        ----------
+        data : dict
+            The trigger homelink for a Tesla vehicle.
+            https://tesla-api.timdorr.com/vehicle/commands/homelink
+        controller : teslajsonpy.Controller
+            The controller that controls updates to the Tesla API.
+
+        Returns
+        -------
+        None
+
+        """
+        super().__init__(data, controller)
+        self.type = "trigger homelink"
+        self.hass_type = "button"
+        self.name = self._name()
+        self.uniq_name = self._uniq_name()
+        self.__longitude = None
+        self.__latitude = None
+        self.__homelink_device_count = None
+        self.__homelink_nearby = None
+        self.__homelink_available = False
+
+    async def async_update(self, wake_if_asleep=False, force=False):
+        """Update the trigger homelink of the vehicle."""
+        await super().async_update(wake_if_asleep=wake_if_asleep)
+        self.refresh()
+
+    @staticmethod
+    def has_battery() -> bool:
+        """Return whether the device has a battery."""
+        return False
+
+    async def trigger_homelink(self) -> None:
+        """Trigger Homelink."""
+        if self.__latitude is not None and self.__longitude is not None:
+            if not self.__homelink_device_count:
+                raise HomelinkError(f"No homelink devices added to {self.car_name()}.")
+            if not self.__homelink_nearby:
+                raise HomelinkError(f"No homelink devices near {self.car_name()}.")
+            await self._controller.api(
+                "TRIGGER_HOMELINK",
+                path_vars={"vehicle_id": self._id},
+                lat=self.__latitude,
+                lon=self.__longitude,
+                wake_if_asleep=True,
+            )
+
+    def refresh(self) -> None:
+        """Refresh data.
+
+        This assumes the controller has already been updated
+        """
+        super().refresh()
+        data = self._controller.get_drive_params(self._id)
+        if data:
+            if data["native_location_supported"]:
+                self.__longitude = data["native_longitude"]
+                self.__latitude = data["native_latitude"]
+            else:
+                self.__longitude = data["longitude"]
+                self.__latitude = data["latitude"]
+        data = self._controller.get_state_params(self._id)
+        if data:
+            self.__homelink_device_count = data["homelink_device_count"]
+            self.__homelink_nearby = data["homelink_nearby"]
+        self.__homelink_available = bool(self.__homelink_device_count)
+
+    def available(self) -> bool:
```

Comment:
Availability in HA normally means it's network accessible too. Should the test also include a device online test?

Reply:
I was looking into this and got stuck, I did discover that when my laptop running the devcontainer went offline, all devices became unavailable except this button, so it should be fixed for sure. However I wasn't able to find out how the other devices determine if they're online.
Any pointers?

Reply:
I think the other entities will set availability false if data doesn't return properly. It's been a while so you may just have to look.

Reply:
ahh the device online test is done through the Entity's available property (in the tesla integration, not teslajsonpy), which I initially overrode and forgot to include it's super value in the comparison. 
Fixed in [7b892e2](https://github.com/alandtse/tesla/pull/168/commits/7b892e229017cd4a84955bc0cf647af13fa91437) and tested by forcing the devcontainer offline.

Reply:
Ok.